### PR TITLE
Fix numerous compiler warnings and errors

### PR DESCRIPTION
**Pull request by jeremyong**
## Description

Depending on OS and compiler, certain types may not be included
transitively by headers that were included previously. This commit
produces a clean build on the latest trunk versions of GCC, Clang,
and QT.

### CODE REVIEW

File: qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp [hunk 1]
```diff
@@ -438,7 +438,7 @@ void PerformanceCounterSelection::Load()
         selectedCounters.insert(m_UuidToCounter[uuid]);
       }
 
-      SetSelectedCounters(selectedCounters.toList());
+      SetSelectedCounters(selectedCounters.values());
```
Comment (baldurk) [inline qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp]: What's the reason for this change? Checking the `QSet` docs, `toList()` is equivalent to `values()` and both have been around since our minspec (Qt 5.6).

File: renderdoc/common/dds_readwrite.h [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #pragma once
 
 #include "api/replay/data_types.h"
+#include <cstdio>
```
Comment (baldurk) [inline renderdoc/common/dds_readwrite.h]: `<stdio.h>` instead of `<cstdio>` please.

**Review by baldurk — CHANGES_REQUESTED** [review]
Thanks for the changes, I've left a couple of notes. Out of interest what platform were you building on where this was an issue?

File: qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp (hunk 1 above)
Comment (jeremyong) [reply qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp]: When building against more recent versions of QT, toList is marked as `deprecated` which will break the build due to `Werror`

Comment (jeremyong) [reply qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp]: This breaks downstream packaged builds for linux distros like Arch for example (or any other system that has a more recent QT installed)

**Comment by jeremyong** [comment]
I noticed this on Arch Linux. I tend to do a lot of bleeding edge development (mainly experimenting with C++20) so I check my code often against gcc9 and clang10.

File: renderdoc/common/dds_readwrite.h (hunk 2 above)
Comment (jeremyong) [reply renderdoc/common/dds_readwrite.h]: done

**Comment by jeremyong** [comment]
Sorry about the force pushes, I didn't realize at first that I had used `cstdio` more than once

File: qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp (hunk 1 above)
Comment (baldurk) [reply qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp]: `Werror` is not enabled in release builds for this exact reason, so that downstream packaged builds with different compiler setups don't get blocked when warnings appear. If any downstream package is having trouble, they should fix that instead by building a release build.

On top of that those Qt deprecation warnings should be disabled. In `qrenderdoc.pro` there's:

```
DEFINES += QT_NO_DEPRECATED_WARNINGS
```

Which I had to add because there was a ridiculous situation where something was deprecated when its replacement was only added in the immediate previous version, leaving it a mess to try and support a large old and new Qt versions.

Comment (jeremyong) [reply qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp]: OK should I revert this then even though the minspec supports it?

Comment (baldurk) [reply qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp]: I'm fine with the change if it's needed for something, but I wanted to know why it was needed first. It seems like things should work with `toList`? Even on a dev build those warnings should be disabled and if they're not then other things would break. If so then all things being equal I'd prefer to revert it.

Comment (jeremyong) [reply qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp]: Reverted. I just checked the release build and indeed, this change specifically isn't needed to fix the build

File: renderdoc/os/posix/linux/linux_stringio.cpp [hunk 3]
```diff
@@ -22,6 +22,7 @@
  * THE SOFTWARE.
  ******************************************************************************/
 
+#include <cctype>
```
Comment (baldurk) [inline renderdoc/os/posix/linux/linux_stringio.cpp]: I didn't comment on each include to avoid spam, but this should similarly be `<ctype.h>`.

File: qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp (hunk 1 above)
Comment (baldurk) [reply qrenderdoc/Windows/Dialogs/PerformanceCounterSelection.cpp]: Does it block a dev build? It shouldn't, because of the `QT_NO_DEPRECATED_WARNINGS`, so if that's a problem then it should still be investigated separately.

File: renderdoc/os/posix/linux/linux_stringio.cpp (hunk 3 above)
Comment (jeremyong) [reply renderdoc/os/posix/linux/linux_stringio.cpp]: Done

**Review by baldurk — APPROVED** [review]

**Comment by jeremyong** [comment]
Thanks Baldur!
